backend/src/services/billing/billing.service.ts

```diff
@@ -0,0 +1,131 @@
+import { TransactionStatus } from '~/common/enums/enums';
+import {
+  TransactionCreateArgumentsDto,
+  TransactionGetAllItemResponseDto,
+  UserDetailsWithMoneyBalanceDto,
+  UserGetResponseWithMoneyBalanceDto,
+} from '~/common/types/types';
+import {
+  stripe as stripeServ,
+  transaction as transactionServ,
+  user as userServ,
+  userDetails as userDetailsServ,
+} from '~/services/services';
+
+type Constructor = {
+  stripeService: typeof stripeServ;
+  transactionService: typeof transactionServ;
+  userService: typeof userServ;
+  userDetailsService: typeof userDetailsServ;
+};
+
+class Billing {
+  #stripeService: typeof stripeServ;
+
+  #transactionService: typeof transactionServ;
+
+  #userService: typeof userServ;
+
+  #userDetailsService: typeof userDetailsServ;
+
+  public constructor({
+    stripeService,
+    transactionService,
+    userService,
+    userDetailsService,
+  }: Constructor) {
+    this.#stripeService = stripeService;
+    this.#transactionService = transactionService;
+    this.#userService = userService;
+    this.#userDetailsService = userDetailsService;
+  }
+
+  public async replenish(
+    userId: number,
+    amountOfMoney: number,
+  ): Promise<UserDetailsWithMoneyBalanceDto | null> {
+    const userWithBalance = await this.#userService.getByIdWithMoneyBalance(
+      userId,
+    );
+
+    if (!userWithBalance) {
+      return null;
+    }
```

Comment:
In my opinion, userWithBalance cannot be null, because the userId comes from `req.user`. And since it's a protected route, someone without a user cannot make a request to that route :thinking:

Reply:
Sounds reasonable 👍

Reply:
Yes it is, but can not I use Postman and add there a user field?

Reply:
Didn't quite understand your question, do you mean someone can make a request with Postman, without a user?
If so, then that request will throw an error, because the route is protected

Reply:
User is provided through `authorizationPlugin` with getting him from signed token,
Soo trying to inject user object inside request sounds like something not good (without valid auth token it would throw an error as Farid mentioned) 😕